app/models/work.rb

```diff
@@ -85,16 +85,25 @@ def administered_by?(user)
   end
 
   class << self
-    def unfinished_works(user)
-      works_by_user_state(user, ["none", "draft", "awaiting_approval"])
+    def unfinished_works(user, search_terms = nil)
+      works = works_by_user_state(user, ["none", "draft", "awaiting_approval"])
+      filter_works(works, search_terms)
     end
 
-    def completed_works(user)
-      works_by_user_state(user, "approved")
+    def completed_works(user, search_terms = nil)
+      works = works_by_user_state(user, "approved")
+      filter_works(works, search_terms)
     end
 
-    def withdrawn_works(user)
-      works_by_user_state(user, "withdrawn")
+    def withdrawn_works(user, search_terms = nil)
+      works = works_by_user_state(user, "withdrawn")
+      filter_works(works, search_terms)
+    end
+
+    def filter_works(works, search_terms)
+      return works if search_terms.nil?
```

Comment:
```suggestion
      return works unless search_terms
```

Reply:
@mccalluc I think better in positive terms and try to avoid unless.  Is there a solid reason for this change?  I would prefer @hectorcorrea's code.

Reply:
Method is now gone.